redis/tests/support/mod.rs

```diff
@@ -98,18 +102,80 @@ impl RedisServer {
                 redis::ConnectionAddr::Unix(PathBuf::from(&path))
             }
         };
-        RedisServer::new_with_addr(addr, None, |cmd| {
-            cmd.spawn()
-                .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
-        })
+        RedisServer::new_with_addr(
+            addr,
+            None,
+            |cmd| {
+                cmd.spawn()
+                    .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
+            },
+            None,
+        )
+    }
+
+    pub fn new_with_modules(modules: Vec<RedisModules>) -> RedisServer {
+        let server_type = ServerType::get_intended();
+        let addr = match server_type {
+            ServerType::Tcp { tls } => {
+                // this is technically a race but we can't do better with
+                // the tools that redis gives us :(
+                let addr = &"127.0.0.1:0".parse::<SocketAddr>().unwrap().into();
+                let socket = Socket::new(Domain::IPV4, Type::STREAM, None).unwrap();
+                socket.set_reuse_address(true).unwrap();
+                socket.bind(addr).unwrap();
+                socket.listen(1).unwrap();
+                let listener = TcpListener::from(socket);
+                let redis_port = listener.local_addr().unwrap().port();
+                if tls {
+                    redis::ConnectionAddr::TcpTls {
+                        host: "127.0.0.1".to_string(),
+                        port: redis_port,
+                        insecure: true,
+                    }
+                } else {
+                    redis::ConnectionAddr::Tcp("127.0.0.1".to_string(), redis_port)
+                }
+            }
+            ServerType::Unix => {
+                let (a, b) = rand::random::<(u64, u64)>();
+                let path = format!("/tmp/redis-rs-test-{}-{}.sock", a, b);
+                redis::ConnectionAddr::Unix(PathBuf::from(&path))
+            }
+        };
+        RedisServer::new_with_addr(
+            addr,
+            None,
+            |cmd| {
+                cmd.spawn()
+                    .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
+            },
+            Some(modules),
+        )
     }
 
     pub fn new_with_addr<F: FnOnce(&mut process::Command) -> process::Child>(
         addr: redis::ConnectionAddr,
         tls_paths: Option<TlsFilePaths>,
         spawner: F,
+        modules: Option<Vec<RedisModules>>,
     ) -> RedisServer {
         let mut redis_cmd = process::Command::new("redis-server");
+
+        // Load Redis Modules
+        // This will take the enum variants and load them from Environment Variables
+        // This method allows for implementation of other Redis Modules in the future
+        if let Some(mods) = modules {
+            for m in mods {
+                match m {
+                    RedisModules::RedisJSON => {
+                        redis_cmd
```

Comment:
I don't think we should just `unwrap()` here. Would it be better to pass `modules` in as a `HashMap` or `&[(RedisModule, &OsStr)]` or whatever that has the path in it?

Reply:
It is loaded from an environment variable, would it be wise to add a few lines in the `Makefile` that check that the env vars are set?

Otherwise I could also set it to use `expect` with an actually helpful message other that "Error: Not Present"

Reply:
Yeah, I'm questioning whether it makes sense to do this as an environment variable as opposed to in the code.

I prefer to avoid panics in most cases, so maybe this method should be made fallible, returning a `Result` with `Err` in the case of a missing environment variable?

Reply:
To hard code it may make life difficult for testing outside of the CI Environment so I definitely think we should keep the Env Var stage since the location of the libs may vary, for example on the CI it is stored at `/tmp/librejson.so` but on my machine it is stored at `/opt/librejson.so`, but on other machines it may be stored at `/usr/lib/librejson.so`, to wrap the `TestContext` generator with a result would make this better but it'd mean shifting around all of the tests to include `.unwrap()`'s in the tests themselves, or we could do this

```rust
pub fn new() {
    with_modules(&[]).unwrap()
}
pub fn with_modules(modules: &[RedisModule]) -> Result<TestContext, Error> {
    // ...
}
```
either way, at some point, an error is being unwrapped

Reply:
Okay, let's keep the environment variable and call `expect()` on it with a nice explanatory message.